models/resourceoperation.go

```diff
@@ -29,46 +29,49 @@ type ResourceOperation struct {
 	isValidated    bool              // internal member used for validation check
 }
 
-// Custom marshaling to make empty strings null
+// MarshalJSON returns a JSON encoded byte representation of the model and performs custom autofill
 func (ro ResourceOperation) MarshalJSON() ([]byte, error) {
 	test := struct {
-		Index          *string           `json:"index,omitempty"`
-		Operation      *string           `json:"operation,omitempty"`
-		Object         *string           `json:"object,omitempty"`
-		DeviceResource *string           `json:"deviceResource,omitempty"`
-		Parameter      *string           `json:"parameter,omitempty"`
-		Resource       *string           `json:"resource,omitempty"`
-		DeviceCommand  *string           `json:"deviceCommand,omitempty"`
-		Secondary      []string          `json:"secondary,omitempty"`
-		Mappings       map[string]string `json:"mappings,omitempty"`
+		Index          string             `json:"index,omitempty"`
+		Operation      string             `json:"operation,omitempty"`
+		Object         string             `json:"object,omitempty"`
+		DeviceResource string             `json:"deviceResource,omitempty"`
+		Parameter      string             `json:"parameter,omitempty"`
+		Resource       string             `json:"resource,omitempty"`
+		DeviceCommand  string             `json:"deviceCommand,omitempty"`
+		Secondary      []string           `json:"secondary,omitempty"`
+		Mappings       *map[string]string `json:"mappings,omitempty"`
 	}{
-		Secondary: ro.Secondary,
-		Mappings:  ro.Mappings,
+		Index:          ro.Index,
+		Operation:      ro.Operation,
+		Object:         ro.Object,
+		DeviceResource: ro.DeviceResource,
+		Parameter:      ro.Parameter,
+		Resource:       ro.Resource,
+		DeviceCommand:  ro.DeviceCommand,
+		Secondary:      ro.Secondary,
+		Mappings:       &ro.Mappings,
 	}
 
-	// Empty strings are null
-	if ro.Index != "" {
-		test.Index = &ro.Index
-	}
-	if ro.Operation != "" {
-		test.Operation = &ro.Operation
+	// Empty maps are nil
+	if len(ro.Mappings) == 0 {
+		test.Mappings = nil
 	}
+
 	if ro.DeviceResource != "" {
-		test.DeviceResource = &ro.DeviceResource
-		test.Object = &ro.DeviceResource
+		test.DeviceResource = ro.DeviceResource
```

Comment:
I think this line can be removed because `test.DeviceResource` is set in initialization at line 48